### PR TITLE
feat: Add analytics events to lineage features and cleanup

### PR DESCRIPTION
**Pull request by danwom**
Add click tracking to
- Tabs on Table page
- Tabs on Dashboard page
- Column Lineage links
LineageList updated to use TableListItem component

Signed-off-by: Daniel Won <dwon@lyft.com>
### Summary of Changes
Added various analytics events and updated LineageList to use the TableListItem component.

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes, including screenshots of any UI changes.
- [ ] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/components/ResourceListItem/TableListItem/index.tsx [hunk 1]
```diff
@@ -29,10 +29,8 @@ class TableListItem extends React.Component<TableListItemProps, {}> {
     );
   };
 
-  generateResourceIconClass = (
-    databaseId: string,
-    resource: ResourceType
-  ): string => `icon resource-icon ${getSourceIconClass(databaseId, resource)}`;
+  generateResourceIconClass = (databaseId: string): string =>
+    `icon resource-icon ${getSourceIconClass(databaseId, ResourceType.table)}`;
```
Comment (Golodhros) [inline amundsen_application/static/js/components/ResourceListItem/TableListItem/index.tsx:33]: Nit: I would extract this function outside the component

File: amundsen_application/static/js/features/ColumnList/ColumnLineage/index.tsx [hunk 2]
```diff
@@ -42,6 +43,18 @@ const getLink = (table) => {
   return `/table_detail/${cluster}/${database}/${schema}/${name}?source=column_lineage`;
 };
 
+const onLineageClick = (e) => {
+  logClick(e, {
+    target_id: 'column_lineage',
+  });
```
Comment (Golodhros) [inline amundsen_application/static/js/features/ColumnList/ColumnLineage/index.tsx:49]: Could we add Redux actions as well?

File: amundsen_application/static/js/features/ColumnList/ColumnLineage/index.tsx [hunk 3]
```diff
@@ -42,6 +43,18 @@ const getLink = (table) => {
...
+};
+
+const onSeeMoreClick = (e) => {
+  logClick(e, {
+    target_id: 'column_lineage_see_more',
+  });
```
Comment (Golodhros) [inline amundsen_application/static/js/features/ColumnList/ColumnLineage/index.tsx:55]: Same here

File: amundsen_application/static/js/pages/DashboardPage/index.tsx [hunk 4]
```diff
@@ -159,7 +159,19 @@ export class DashboardPage extends React.Component<
...
+    return (
+      <TabsComponent
+        tabs={tabInfo}
+        defaultTab="tables"
+        onSelect={(key) => {
+          logAction({
+            command: 'click',
+            target_id: 'dashboard_page_tab',
+            label: key,
+          });
```
Comment (Golodhros) [inline amundsen_application/static/js/pages/DashboardPage/index.tsx:171]: Same, could we add a redux action?

File: amundsen_application/static/js/pages/TableDetailPage/index.tsx [hunk 5]
```diff
@@ -299,6 +299,11 @@ export class TableDetail extends React.Component<
         defaultTab={Constants.COLUMN_TAB_KEY}
         onSelect={(key) => {
           this.setState({ currentTab: key });
+          logAction({
+            command: 'click',
+            target_id: 'table_detail_tab',
+            label: key,
+          });
```
Comment (Golodhros) [inline amundsen_application/static/js/pages/TableDetailPage/index.tsx:306]: Could we add a redux action?

**Review by Golodhros — CHANGES_REQUESTED** [review]
Looks good, but I am missing the redux actions that will replace the use of logClick.
Without them, this code is already legacy code!

**Comment by dorianj** [comment]
Hey @danwom, if you could try to get this landed in the next few days, I'll grab it as part of the monorepo switchover.

**Review by Golodhros — APPROVED** [review]
Let's clean the tech debt after the monorepo move.

**Comment by danwom** [comment]
From what I understand we have `analyticsMiddleware` which listens to every fired redux action and checks for a `meta.analytics` field. These analytics events get picked up and fired through an analytics instance that should handle multiple providers.

I think piggybacking on existing events and tracking those actions is great, but events that don't affect changes on the global scope such as clicking on an external link seemed to raise concern. I tried a few different ways to create redux actions for each trackable event, but it got pretty messy with a lot of bloat that I wasn't really happy with. Maybe I'm missing some obvious shortcut or simplification.

This probably needs a bigger discussion and a better plan but I took some intermediary step by moving `logClick` and `moveAction` into the `analytics` file and calling `trackEvent` in the `logAction`. Right now `logAction` uses the analytics provider as well as the built in `/api/log/v0/log_event` API. As a next step we could add this API as another analytics provider.

File: amundsen_application/static/js/features/ColumnList/ColumnLineage/index.tsx [hunk 6]
```diff
@@ -42,6 +43,18 @@ const getLink = (table) => {
   return `/table_detail/${cluster}/${database}/${schema}/${name}?source=column_lineage`;
 };
 
+const onLineageClick = (e) => {
+  logClick(e, {
```
Comment (Golodhros) [inline amundsen_application/static/js/features/ColumnList/ColumnLineage/index.tsx:47]: Nit: On the internal handlers, we use the "handle" prefix, while on the component API we use 'on' as the prefix. You can check more in https://github.com/amundsen-io/amundsenfrontendlibrary/blob/master/docs/react_naming_conventions.md#handler-functions

**Review by Golodhros — APPROVED** [review]
I understand the analytics tracking via Redux actions is not easy to grab at first, but we decided that was the cleaner and more extensible way of tracking for different providers.

I can take a swing on this one later so we align on how to do it.

**Pull request merged**